Add introduction doc (#459)
oam-dev/kubevela: docs/introduction.md

# Introduction to KubeVela

![alt](../resources/KubeVela-01.png)

> Welcome to KubeVela! This documentation covers why KubeVela, what KubeVela is, and how KubeVela vs. others.

Reviewer: ```suggestion
Welcome to KubeVela! This documentation covers why we build KubeVela, what KubeVela is, and comparison of KubeVela vs. others.
```

## Why KubeVela?

> The modern patterns of cloud-native benefits application management a lot with consistency by abstracting away the details of clouds and infrastructures with Kubernetes. But this also introduced another layer of complexity: pods, containers, proxies, claims, service meshes and more. We've seen lack of understanding of Kubernetes and its learning curve impacted user experience, slowed down the software shipping, created misconfiguration or even failures. Developers become doubtful to the value of cloud-native revolution: "why bother me with those YAML files?" they asked. 

Reviewer: ```suggestion
The modern patterns of cloud-native application management has been merited with consistency across clouds and infrastructure by abstracting away the low level details with Kubernetes. However, this also introduced another layer of complexity: pods, ingresses, claims, service meshes, and so on. So far, the lack of Kubernetes knowledge due to its high learning curve has impacted user experience, slowed down the software shipping, led to user misconfiguration and production issues. Developers start questioning the value of cloud-native revolution, "why bother me with those YAML files?".
```

> Another challenge we saw is during solving this problem. There are already many successful platforms (including PaaS and Serverless) in the community that bring great developer experience to cloud-native world. But for platform engineers in many organizations, this poses a new challenge in how to adapt or extend these platforms to meet their own requirements in various scenarios. Essentially, platforms aiming at fixing developer facing issues tend to be highly opinionated in terms of user interfaces, assumptions and implementations. This makes extending such platforms with wider ecosystem almost impossible or at least very difficult, and many of them eventually become close systems.

Reviewer: ```suggestion

Another challenge we have seen: There are many successful platforms (including PaaS and Serverless) in the community that provide great user experience. But for platform engineers in many organizations, this poses a new challenge in how to adapt to or extend these platforms to meet their in-house requirements and scenarios. Essentially, platforms aimed at resolving developer issues tend to be highly opinionated in what should be exposed. This makes extending such platforms within the community ecosystem almost impossible or at least very difficult, and many of them eventually become close systems.
```

KubeVela intends to make shipping applications more enjoyable for developers, and provides platform builders with ecosystem native extensibility to create their own platforms on top of.

## What is KubeVela?

> *For developers, KubeVela is a easy-to-use tool that enables developers to describe and ship their applications with simple commands.*

Reviewer: ```suggestion
*For developers, KubeVela is an easy-to-use tool that enables developers to describe and ship their applications.*
```

In this context, KubeVela provides ease of use on top of Kubernetes to enable developers ship their softwares with efficiency and confidence to any cluster. No more API objects, no more container patterns. This workflow could be easily automated by CI/CD pipelines.

> *For platform engineers, KubeVela is a fully extensible engine that allows platform builders to create something more complex on top of, with easy and Kubernetes native approach.*

Reviewer: ```suggestion
*For platform engineers, KubeVela is an extensible engine where platform builders can build more abstractions in a simplified and Kubernetes native approach.*
```
